### PR TITLE
FEAT - Split Lab build into an action of its own

### LINKED ISSUE #27: ENH -  Pull JupyterLab build and install steps into reusable action
Currently there are a few steps in .github/actions/jlab-a11y-test/action.yml but they could be pulled out into a separate action.

### PR DESCRIPTION
**Pull request by trallard**
Closes #27 

## TODO as part of this PR

- [x] It should be updated once #29  is merged into `main`
- [x] Add option to use an `SHA` for Lumino build/link in case this is needed
- [x] Update comments and docs to match the changes

🎉 I now modified the PR so that we can take the repo and `ref` of almost any package to build and test with JupyterLab

This run combines `jupyterlab/jupyterlab` with the branch from mark's lumino's PR that @gabalafou  is currently reviewing (`scmmmh/lumino`) https://github.com/Quansight-Labs/jupyter-a11y-testing/actions/runs/3961294989

### CODE REVIEW

**Comment by trallard** [comment]
That summary formatting is wonky, so I might need to figure out a way to deal with it... not urgent tho

File: .github/actions/link-package/action.yml
Comment (gabalafou) [inline .github/actions/link-package/action.yml]: Do we need to add some note somewhere in this file that this action needs to be run in a certain context, like within a workflow that sets up `yarn` and `conda`? I mean, like, some way to indicate that this is very much not a standalone action.

File: .github/actions/link-package/action.yml
Comment (gabalafou) [inline .github/actions/link-package/action.yml:51]: Does this work? I'm not sure but I think you actually have to go through each of the (sub)packages in the Lumino monorepo and link them one by one. 

Is that what your "todo" comment is about?

File: .github/workflows/accessibility-test-jupyterlab.yml
Comment (gabalafou) [inline .github/workflows/accessibility-test-jupyterlab.yml]: This description is not the same as the other 😛

**Review by gabalafou — APPROVED** [review]

**Comment by gabalafou** [comment]
Approved for now. If anything's not working we can fix it in subsequent pull requests.

**Comment by trallard** [comment]
I am going to merge this and if needed work on smaller self-contained improvements

File: .github/actions/link-package/action.yml
Comment (trallard) [reply .github/actions/link-package/action.yml:51]: This works as is - will make improvements for robustness in a later iteration

Comment (gabalafou) [reply .github/actions/link-package/action.yml:51]: Yeah, I remember we chatted about this with @steff456, I think, and I may have just been flat out wrong about needing to link every sub-package versus the top-level package.

File: .github/actions/link-package/action.yml
Comment (trallard) [reply .github/actions/link-package/action.yml]: Added now

**Comment by trallard** [comment]
Note I had to upgrade to node >= 18 on our deps (ljlab) since now whatever is in `master` dropped support for node16

File: .github/actions/link-package/action.yml
Comment (gabalafou) [inline .github/actions/link-package/action.yml:43]: `YARN_IGNORE_PATH` 😵‍💫? Why not use `jlpm link` here?

File: .github/actions/link-package/action.yml
Comment (gabalafou) [inline .github/actions/link-package/action.yml]:
```suggestion
        echo "::endgroup::"
```

File: .github/workflows/accessibility-test-jupyterlab.yml
Comment (gabalafou) [inline .github/workflows/accessibility-test-jupyterlab.yml]: This feels like a minor thing, but I think "JLab testing repository" risks creating confusion than just "repository" (which I think conventionally in GitHub Actions means "**this** repository")

File: README.md
Comment (gabalafou) [inline README.md]:
```suggestion
│       ├── testing/jupyterlab/manual-testing-scripts   # "recipes" that explain in plain language how automated tests can be also be carried out manually
```

Comment (gabalafou) [reply README.md]: This makes me think there's something wrong with the naming or information hierarchy. This is why I originally named the directory something like test-info-pages... In my mental model, every accessibility test in this repo should come with its own human-readable page that explains what the test is, why it matters, how to carry out the test manually, and maybe other info.

File: README.md
Comment (gabalafou) [inline README.md]:
```suggestion
├── .pre-commit-config.yaml                             # configuration file for the pre-commit hooks
```

File: README.md
Comment (gabalafou) [inline README.md]:
```suggestion
│   └── testing/scripts                                 # set of manual testing scripts for JupyterLab (include relevant WCAG success criteria and step-by-step guides to audit)
```

File: README.md
Comment (gabalafou) [inline README.md]:
```suggestion
│   └── testing/notebooks                               # set of reference Jupyter notebooks to be used in the automated tests
```

File: README.md
Comment (gabalafou) [reply README.md]: We should probably delete `testing/scripts` lest it cause confusion with `testing/jupyterlab/manual-testing-scripts`.

Or we need to decide on the path forward for this repo and our work.

**Review by gabalafou — CHANGES_REQUESTED** [review]
I **love** that we are going to have a way to build a specific version of Lumino into JupyterLab with these changes.

I caught a couple small errors that should probably be fixed before merging.

I also have some comments that shouldn't block this.

File: .github/actions/link-package/action.yml
Comment (trallard) [reply .github/actions/link-package/action.yml:43]: I was using `jlpm link` but started failing with

```
ERROR conda.cli.main_run:execute(47): `conda run jlpm link` failed. (See above for error)
Unknown Syntax Error: Not enough positional arguments.

$ yarn link [-A,--all] [-p,--private] [-r,--relative] <destination>
```

 see https://github.com/Quansight-Labs/jupyter-a11y-testing/actions/runs/4439607697/jobs/7792297908#step:4:4042

I tried multiple things, and the only thing that worked was adding that bit 🤷🏽‍♀️ 
see https://classic.yarnpkg.com/lang/en/docs/cli/link/

we can try and dig the bug now or merge and check later - but I am also not sure if this happened due to some changes in lumino itself

File: .github/workflows/accessibility-test-jupyterlab.yml
Comment (trallard) [reply .github/workflows/accessibility-test-jupyterlab.yml]: Ok will revert then

File: README.md
Comment (trallard) [reply README.md]: yes we need to discuss this on Monday I once again was not sure when creating the tree for the repo - will leave it as is now and we can make a decision on Monday

**Comment by trallard** [comment]
@gabalafou fixed the issues - except for the `jlpm link` bit

**Pull request merged**
